src/api.ts

```diff
@@ -22,42 +22,109 @@ export enum PluginType {
   PLATFORM = "platform",
 }
 
+/**
+ * The {PluginInitializer} is a method which must be the default export for every homebridge plugin.
+ * It is called once the plugin is loaded from disk.
+ */
 export interface PluginInitializer {
 
+  /**
+   * When the initializer is called the plugin must use the provided api instance and call the appropriate
+   * register methods - {@link API.registerAccessory} or {@link API.registerPlatform} - in order to
+   * correctly register for the following startup sequence.
+   *
+   * @param {API} api
+   */
   (api: API): void;
 
 }
 
 export interface AccessoryPluginConstructor {
-  new(logger: Logging, config: AccessoryConfig): AccessoryPlugin;
+  new(logger: Logging, config: AccessoryConfig, api: API): AccessoryPlugin;
 }
 
 export interface AccessoryPlugin {
 
+  /**
+   * Optional method which will be called if a 'identify' of a Accessory is requested by HomeKit.
+   */
   identify?(): void;
 
+  /**
+   * This method will be called once on startup to query all services to be exposed by the Accessory.
+   * All event handlers for characteristics should be set up before the array is returned.
+   *
+   * @returns {Service[]} services - returned services will be added to the Accessory
+   */
   getServices(): Service[];
 
 }
 
 export interface PlatformPluginConstructor {
-  new(logger: Logging, config: PlatformConfig, api: API): PlatformPlugin;
+  new(logger: Logging, config: PlatformConfig, api: API): DynamicPlatformPlugin | StaticPlatformPlugin | IndependentPlatformPlugin;
 }
 
-export interface PlatformPlugin { // also referred to as "dynamic" platform plugin
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface PlatformPlugin {} // not exported to the public in index.ts
+
+/**
+ * Platform that is able to dynamically add or remove accessories.
+ * All configured accessories are stored to disk and recreated on startup.
+ * Accessories can be added or removed by using {@link API.registerPlatformAccessories} or {@link API.unregisterPlatformAccessories}.
+ */
+export interface DynamicPlatformPlugin extends PlatformPlugin {
 
+  /**
+   * This method is called for every PlatformAccessory, which is recreated from disk on startup.
+   * It should be used to properly initialize the Accessory and setup all event handlers for
+   * all services and their characteristics.
+   *
+   * @param {PlatformAccessory} accessory which needs to be configured
+   */
   configureAccessory(accessory: PlatformAccessory): void;
 
 }
 
-export interface LegacyPlatformPlugin {
+/**
+ * Platform that exposes all available characteristics at the start of the plugin.
+ * The set of accessories can not change at runtime.
+ * The bridge waits for all callbacks to return before it is published and accessible by HomeKit controllers.
+ */
+export interface StaticPlatformPlugin extends PlatformPlugin {
```

Comment:
🙌🏻

Reply:
This is the interface you were searching for? This PR also fixes the return type of the PlatformPluginConstructor type, which previously only allowed to return a dynamic platform 😅

Reply:
Yup, this should mix the type error that I get now. I'll keep an eye on NPM for the new beta with these changes.

Reply:
beta 45 should be released